cmake/modules/CompileWarnings.cmake

```diff
@@ -2,7 +2,7 @@ macro(wpilib_target_warnings target)
     if(NOT MSVC)
         target_compile_options(${target} PRIVATE -Wall -pedantic -Wextra -Werror -Wno-unused-parameter ${WPILIB_TARGET_WARNINGS})
     else()
-        target_compile_options(${target} PRIVATE /wd4146 /wd4244 /wd4251 /wd4267 /WX /D_CRT_SECURE_NO_WARNINGS ${WPILIB_TARGET_WARNINGS})
```

Comment:
If there are errors we should either fix them or suppress the specific warnings.

Reply:
The error was
```
D:\a\allwpilib\allwpilib\build\_deps\apriltaglib-src\common/math_util.h(52,1): error C2220: the following warning is treated as an error [D:\a\allwpilib\allwpilib\build\apriltag\apriltag.vcxproj]
D:\a\allwpilib\allwpilib\build\_deps\apriltaglib-src\common/math_util.h(52,1): warning C4005: 'max': macro redefinition [D:\a\allwpilib\allwpilib\build\apriltag\apriltag.vcxproj]
  Drive.cpp
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\ucrt\stdlib.h(1282,1): message : see previous definition of 'max' [D:\a\allwpilib\allwpilib\build\apriltag\apriltag.vcxproj]
D:\a\allwpilib\allwpilib\build\_deps\apriltaglib-src\common/math_util.h(53,1): warning C4005: 'min': macro redefinition [D:\a\allwpilib\allwpilib\build\apriltag\apriltag.vcxproj]
C:\Program Files (x86)\Windows Kits\10\Include\10.0.22621.0\ucrt\stdlib.h(1283,1): message : see previous definition of 'min' [D:\a\allwpilib\allwpilib\build\apriltag\apriltag.vcxproj]
```
I disabled `/WX` to see if I can get a working windows build via cmake at all :)

Reply:
@rzblue is this also done in the gradle build?

Reply:
I'm not sure if it's enabled for apriltaglib, but we do use `/WX` for gradle:
https://github.com/wpilibsuite/native-utils/blob/3daaca03995c44838bb553b883f0809821103a44/src/main/java/edu/wpi/first/nativeutils/WPINativeUtilsExtension.java#L51